core/services/ocr2/plugins/functions/plugin.go

```diff
@@ -113,3 +154,27 @@ func NewConnector(gwcCfg *connector.ConnectorConfig, ethKeystore keystore.Eth, c
 	handler.SetConnector(connector)
 	return connector, nil
 }
+
+func getDecryptionQueueConfig(pluginConfig config.PluginConfig) config.DecryptionQueueConfig {
+	dqc := config.DecryptionQueueConfig{
+		MaxDecryptionQueueLength:           100,
+		MaxCiphertextBytes:                 10_000,
+		MaxCiphertextIdLength:              100,
+		CompletedDecryptionCacheTimeoutSec: 300,
+	}
+
+	if pluginConfig.MaxDecryptionQueueLength > 0 {
```

Comment:
Those validations should go into ValidatePluginConfig() in config.go (which currently does nothing..) When you move Threshold config to a nested struct that can be nil, you can validate values only if they exist and skip otherwise.

You can also add the defaults over there as consts or maybe better require all values to be set explicitly?

Reply:
I prefer default values at least for now, then maybe we can deprecate if we decide we don't want defaults.